cmake/multi_image.cmake

```diff
@@ -125,6 +125,82 @@ function(add_child_image_from_source)
   set(${ACI_NAME}_BOARD ${ACI_BOARD})
 
   message("\n=== child image ${ACI_NAME} - ${ACI_DOMAIN}${inherited} begin ===")
+
+  # Check if the folder 'child_image/<ACI_NAME>' exists, and apply all
+  # relevant kconfig sources.
+  #
+  # <sample> DIRECTORY
+  # | - prj.conf (A)
+  # | - prj_<desc>.conf (B)
+  # | - boards DIRECTORY
+  # | | - <board>.conf (C)
+  # | | - <board>_<desc>.conf (D)
+  # | - child_image DIRECTORY
+  # | | <ACI_NAME> DIRECTORY
+  # | | - boards DIRECTORY
+  # | | | - <board>.conf (E)        If present, use instead of (C)
+  # | | | - <board>_<desc>.conf (F) If present, use instead of (D)
+  # | | - prj.conf (G)              If present, use instead of (A)
+  # | |                             Note that (C) is ignored if this is present.
+  # | |                             Use (E) instead.
+  # | | - prj_<desc>.conf (H)       If present, used instead of (B) when user
+  # | |                             specify `-DCONF_FILE=prj_<desc>.conf for
+  # | |                             parent image. Note that any (C) is ignored
+  # | |                             if this is present. Use (F) instead.
+  # | | - <ACI_name>_prj.conf (I)   Fragment, used together with (A) and (C)
+  # | | - <ACI_name>_prj_<desc>.conf (J) Fragment, used together with (B) and (D)
+
+  set(alt_conf ${APPLICATION_SOURCE_DIR}/child_image/${ACI_NAME})
+  if (EXISTS ${alt_conf})
+    # First of all we need to clean up the 'CONF_FILE' variable. It could be
+    # a list if (C) or (D) is present. If (C) or (D) is present and
+    # (G) or (H) is present (C) or (D)  will not be used. Hence for the purpose
+    # of these checks, we remove the (C) or (D) entry from  the CONF_FILE
+    # variable if it exists.
+    list(LENGTH CONF_FILE CONF_FILE_LENGTH)
+    if (NOT ${CONF_FILE_LENGTH} EQUAL 1)
+      list(GET CONF_FILE_AS_LIST 0 safe_conf_file)
+    else()
+      set(safe_conf_file ${CONF_FILE})
+    endif()
+
+    # Now that we have a conf file variable which we know only contain a single
+    # entry, identify if any special configuration variant is used.
+    # These are the prj_<alternative name>.conf variants.
+    if (${safe_conf_file} MATCHES "prj_(.*).conf" AND
+         (NOT ${CMAKE_MATCH_1} STREQUAL ${BOARD}))
+      set(prj_conf_file_name prj_${CMAKE_MATCH_1}.conf)
+      set(board_conf_file_name ${ACI_BOARD}_${CMAKE_MATCH_1}.conf)
+    else()
+      set(prj_conf_file_name prj.conf)
+      set(board_conf_file_name ${ACI_BOARD}.conf)
+    endif()
```

Comment:
Part of should be reworked when #3569 completes, as that will provide `zephyr_file()` and include board revision feature.

Reply:
done